File: OCaml/src/Lib/Math/src/MathTest.ml
open OUnit;;
open Math;;

module MathTest =

 struct

  let ae = assert_equal;;

  let test_fib _ =
   assert_equal 0 (Math.fib 0);
   assert_equal 1 (Math.fib 1);
   assert_equal 6765 (Math.fib 20);; 

  let test_fact _ =
   assert_equal 1 (Math.fact 0);
   assert_equal 120 (Math.fact 5);;

  let test_times _ =
   assert_equal 0 (Math.times 0 5);
   assert_equal 0 (Math.times 5 0);
   assert_equal 50 (Math.times 5 10);;

  let test_double _ =
   assert_equal 10 (Math.double 5);;

  let test_induction_fact _ =
   assert_equal 1 (Math.induction_fact 0);
   ae 120 (Math.induction_fact 5);;

  let test_induction_sum_int _ =
   ae 0 (Math.induction_sum_int 0);
   ae 1 (Math.induction_sum_int 1);
   ae 210 (Math.induction_sum_int 20);;

  let test_induction_sum_sqr _ =
   ae 0 (Math.induction_sum_sqr 0);
   ae 1 (Math.induction_sum_sqr 1);
   ae 2870 (Math.induction_sum_sqr 20);;

  let suite = "OUnit" >:::
   [
    "test_fib" >:: test_fib;
    "test_fact" >:: test_fact;
    "test_times" >:: test_times;
    "test_double" >:: test_double;
    "test_induction_fact" >:: test_induction_fact;
    "test_induction_sum_int" >:: test_induction_sum_int;
    "test_induction_sum_sqr" >:: test_induction_sum_sqr;
   ];;

 end;;

File: Go/src/Lib/Math/math_test.go
package math

import (
 "testing"
)

func try(t *testing.T, b bool, m string) {
 if b == false {
  t.Error("Failure: " + m)
 } else {
  t.Log("Success: " + m)
 }
}


func Test_Fib(t *testing.T) {
 try(t, Fib(0) == 0, "fib(0) == 0")
 try(t, Fib(1) == 1, "fib(1) == 1")
 try(t, Fib(20) == 6765, "fib(20) == 6765")
}


func Test_Fact(t *testing.T) {
 try(t, Fact(0) == 1, "fact(0) == 1")
 try(t, Fact(5) == 120, "fact(5) == 120")
}


func Test_Times(t *testing.T) {
 try(t, Times(0, 5) == 0, "times(0, 5) == 0")
 try(t, Times(5, 0) == 0, "times(5, 0) == 0")
 try(t, Times(5, 10) == 50, "times(5, 10) == 50")
}


func Test_Double(t *testing.T) {
 try(t, Double(5) == 10, "double(5) == 10")
}


func Test_Induction_Fact(t *testing.T) {
 try(t, Induction_Fact(0) == Fact(0), "induction_fact(0) == fact(0)")
 try(t, Induction_Fact(1) == Fact(1), "induction_fact(1) == fact(1)")
 try(t, Induction_Fact(5) == Fact(5), "induction_fact(5) == fact(5)")
}

func Test_Induction_Sum_Int(t *testing.T) {
 try(t, Induction_Sum_Int(0) == 0, "induction_sum_int(0) == 0")
 try(t, Induction_Sum_Int(1) == 1, "induction_sum_int(1) == 1")
 try(t, Induction_Sum_Int(20) == 210, "induction_sum_int(20) == 210")
}

func Test_Induction_Sum_Sqr(t *testing.T) {
 try(t, Induction_Sum_Sqr(0) == 0, "induction_sum_sqr(0) == 0")
 try(t, Induction_Sum_Sqr(1) == 1, "induction_sum_sqr(1) == 1")
 try(t, Induction_Sum_Sqr(20) == 2870, "induction_sum_sqr(20) == 2870")
}
